util/build_configs/cray-internal/generate-modulefile.bash

```diff
@@ -72,10 +72,43 @@ conflicts xt-mpich2 < 6.0.0
 conflict chapel
 #conflict gcc
 
+##
+## Now, we're going to attempt to get to the right build type
+###
+#
+set network seastar
+if { [ info exists env(XTPE_NETWORK_TARGET) ] } {
+    set network $env(XTPE_NETWORK_TARGET)
+} elseif { [ info exists env(CRAYPE_NETWORK_TARGET) ] } {
+    set network $env(CRAYPE_NETWORK_TARGET)
+}
+set accel 0
+if { [ info exists env(CRAY_ACCEL_TARGET) ] } {
+    set accel 1
+}
+
+
+setenv MPICH_GNI_DYNAMIC_CONN disabled
+if { [string match gemini $network] } {
+    set CHPL_HOST_PLATFORM cray-xe
+} elseif { [string match aries $network] } {
+    set CHPL_HOST_PLATFORM cray-xc
+} elseif { [string match ofi $network] } {
+    set CHPL_HOST_PLATFORM cray-cs
+    if { [file exists /etc/opt/cray/release/cray-release] } {
+        set CRAY_REL_INFO [exec cat /etc/opt/cray/release/cray-release]
+        if { [string match "PRODUCT=*Shasta*" $CRAY_REL_INFO] } {
+            set CHPL_HOST_PLATFORM cray-shasta
+        }
```

Comment:
Have we checked with PE groups to see if this is the "right" way to detect shasta?

Reply:
No, it's just what I found on the systems.  (It would be the OS group to check with, actually.  I'll do that.)